bin/create_folders.py

```diff
@@ -47,21 +58,22 @@ def prefix_filename(
 
     """
     filename = old_file_entity.name
-    predictions_file_name = f"{prefix_name}_{filename}"
+    prefixed_filename = f"{prefix_name}_{filename}"
 
     updated_file_entity = synapseutils.changeFileMetaData(
         syn,
         entity=old_file_entity,
-        downloadAs=predictions_file_name,
+        downloadAs=prefixed_filename,
         forceVersion=False,
     )
-    updated_file_entity.name = predictions_file_name
+    updated_file_entity.name = prefixed_filename
     syn.store(updated_file_entity)
```

Comment:
If you are on the 4.1 or later of the Synapse python client - You do not need to do this anymore. changeFileMetaData has a name parameter you can update the entity name with:
https://sagebionetworks.jira.com/browse/SYNPY-448

Reply:
The task that calls `create_folders` runs in a container that has `synapseclient==4.0.0` so I don't think this would work until we update the container again - but I can organize a "nf-synapse-challenge cleanup" Epic to tackle updates to the containers and updated functionality like this, and also include your OOP refactoring work.

Reply:
Can you just change the `CREATE_FOLDERS` `container` to `sagebionetworks/synapsepythonclient:v4.1.1`?

Reply:
My reasoning for not doing it in this PR is that it feels sloppy to update only the `CREATE_FOLDERS` container and not the other ones that use `synapseclient`, and then have to update the logic to reflect `4.1.1` which starts to become scope creep, but I can do this here if that is everyone's preference.

Reply:
I don't think it's a big deal to change just one of the containers this time. They are all getting pulled separately when this runs on Tower anyway